pkg/controller/vsphere/reconciler.go

```diff
@@ -171,7 +199,11 @@ func (r *Reconciler) create() error {
 			return fmt.Errorf("failed to check task status: %w", err)
 		}
 	} else if !taskIsFinished {
-		return fmt.Errorf("%v task %v has not finished", moTask.Info.DescriptionId, moTask.Reference().Value)
+		if taskIsFinished {
```

Comment:
These two statements seem contradictory next to each other? Can this error ever return? Is this the correct way round?